src/shogun/kernel/GaussianKernel.cpp

```diff
@@ -171,15 +126,33 @@ SGMatrix<float64_t> CGaussianKernel::get_parameter_gradient(
 	}
 }
 
-void CGaussianKernel::init()
+float64_t CGaussianKernel::compute(int32_t idx_a, int32_t idx_b)
 {
-	set_width(1.0);
-	sq_lhs=NULL;
-	sq_rhs=NULL;
-	SG_ADD(&m_log_width, "log_width", "Kernel width in log domain", MS_AVAILABLE, GRADIENT_AVAILABLE);
+    float64_t result=distance(idx_a, idx_b);
+    return CMath::exp(-result);
 }
 
-float64_t CGaussianKernel::distance(int32_t idx_a, int32_t idx_b)
+void CGaussianKernel::load_serializable_post() throw (ShogunException)
 {
-	return (sq_lhs[idx_a]+sq_rhs[idx_b]-2*CDotKernel::compute(idx_a,idx_b))/get_width();
+	CKernel::load_serializable_post();
+	if (lhs && rhs)
+		m_distance->init(lhs, rhs);
+}
+
+float64_t CGaussianKernel::distance(int32_t idx_a, int32_t idx_b) const
+{
+	return CShiftInvariantKernel::distance(idx_a, idx_b)/get_width();
```

Comment:
mmm will clang/gcc in `-O3` optimize this into `1/get_width()` being a const and just do a `CShiftInvariantKernel::distance(idx_a, idx_b)*<that const>`?